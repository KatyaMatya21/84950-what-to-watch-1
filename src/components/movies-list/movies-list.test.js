import React from 'react';
import renderer from 'react-test-renderer';

import MoviesList from '../movies-list/movies-list.jsx';

const movies = [
  {
    'id': `1`,
    'title': `What We Do in the Shadows`,
    'src': `img/what-we-do-in-the-shadows.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `2`,
    'title': `What We Do in the Shadows`,
    'src': `img/what-we-do-in-the-shadows.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    'id': `3`,
    'title': `What We Do in the Shadows`,
    'src': `img/what-we-do-in-the-shadows.jpg`,
    'link': `movie-page.html`,
    'preview': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

describe(`MoviesList component`, () => {

  it(`Correctly renders`, () => {
    const MoviesListComponent = renderer
      .create(
          <MoviesList movies={movies} />,
          {
            createNodeMock: (element) => {
              if (element.type === `video`) {
                return {'current': {}};
              }
              return null;
            }
          }
      )
      .toJSON();
    expect(MoviesListComponent).toMatchSnapshot();
  });

});
