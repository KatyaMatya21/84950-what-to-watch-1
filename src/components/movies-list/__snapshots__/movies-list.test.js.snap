// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component Correctly renders 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/macbeth.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/we-need-to-talk-about-kevin.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseOut={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height="175"
        muted={true}
        poster="img/what-we-do-in-the-shadows.jpg"
        width="280"
      >
        <source
          src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        />
      </video>
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
</div>
`;
